### PR TITLE
Avoid invalid NaN lint machine-applicable suggestion in const context

### LINKED ISSUE #114471: `invalid_nan_comparisons` suggests invalid  `5f32.is_nan()` in const context
### Code

```Rust
fn main() {
    const TEST: bool = 5f32 == f32::NAN;
}
```

### Current output

```Shell
warning: constant `TEST` is never used
 --> src/main.rs:2:11
  |
2 |     const TEST: bool = 5f32 == f32::NAN;
  |           ^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: incorrect NaN comparison, NaN cannot be directly compared to itself
 --> src/main.rs:2:24
  |
2 |     const TEST: bool = 5f32 == f32::NAN;
  |                        ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(invalid_nan_comparisons)]` on by default
help: use `f32::is_nan()` or `f64::is_nan()` instead
  |
2 -     const TEST: bool = 5f32 == f32::NAN;
2 +     const TEST: bool = 5f32.is_nan();
  |

warning: 2 warnings emitted
```

### Desired output

_No response_

### Rationale and extra context

The suggestion is auto applicable
```rust
fn main() {
    const TEST: bool = 5f32.is_nan();
}
```

but does not compiled because it relies on unstabilized features
```
error: `core::f32::<impl f32>::is_nan` is not yet stable as a const fn
 --> src/main.rs:2:24
  |
2 |     const TEST: bool = 5f32.is_nan();
  |                        ^^^^^^^^^^^^^
  |
  = help: add `#![feature(const_float_classify)]` to the crate attributes to enable

error: aborting due to previous error
```

### Other cases

_No response_

### Anything else?

rustc 1.73.0-nightly (474709a9a 2023-08-03)
binary: rustc
commit-hash: 474709a9a2a74a8bcf0055fadb335d0ca0d2d939
commit-date: 2023-08-03
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5

### PR DESCRIPTION
**Pull request by Urgau**
This PR removes the machine-applicable suggestion in const context for the `invalid_nan_comparision` lint ~~and replace it with a simple help~~.

Fixes https://github.com/rust-lang/rust/issues/114471

### CODE REVIEW

**Comment by compiler-errors** [comment]
I wonder if we should just see what's blocking stabilizing `is_nan` in const contexts :laughing:

**Comment by Urgau** [comment]
> I wonder if we should just see what's blocking stabilizing `is_nan` in const contexts laughing

The tracking issue https://github.com/rust-lang/rust/issues/72505, say that it's blocked on https://github.com/rust-lang/rust/issues/57241, ie general availability of floating point types in `const fn`.

**Comment by compiler-errors** [comment]
Well in that case, shouldn't we just be suggesting *nothing*? I don't think turning this into a help that still tells folks to use the same fn is the right solution.

**Comment by Urgau** [comment]
It's a bit unsatisfying to not have any suggestion for const context, but I agree, better suggested nothing than suggested functions that have no plan to stabilization in the short or medium term.

Updated the PR to completely remove the suggestion.

File: compiler/rustc_lint/src/types.rs [hunk 1]
```diff
@@ -572,32 +572,35 @@ fn lint_nan<'tcx>(
...
 
     fn eq_ne(
+        cx: &LateContext<'_>,
         e: &hir::Expr<'_>,
         l: &hir::Expr<'_>,
         r: &hir::Expr<'_>,
         f: impl FnOnce(Span, Span) -> InvalidNanComparisonsSuggestion,
     ) -> InvalidNanComparisons {
-        let suggestion =
+        let suggestion = (!cx.tcx.hir().is_inside_const_context(e.hir_id)).then(|| {
```
Comment (compiler-errors) [inline compiler/rustc_lint/src/types.rs:581]: nit: this is a somewhat confusing way of avoiding an `if` statement :sweat_smile:

Comment (compiler-errors) [inline compiler/rustc_lint/src/types.rs:581]:
```suggestion
        // FIXME(#72505): This suggestion can be restored if `f{32,64}::is_nan` is made const.
        let suggestion = (!cx.tcx.hir().is_inside_const_context(e.hir_id)).then(|| {
```

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**